[Bug]: Addon build fails due to missing `@storybook/manager/globals` on 8.2.x

### Describe the bug

After upgrading a fresh addon-kit generated project's dependencies to 8.2.x, the `npm run build` command fails with:
 
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/tomek/Sites/tmp/addon-kit/node_modules/@storybook/manager/globals' imported from /Users/tomek/Sites/tmp/addon-kit/tsup.config.bundled_4chh9op0kq8.mjs
```

Looks like in 8.2 the `@storybook/manager` includes only a single file (package.json).

See the recording below for comparison between `@storybook/manager@8.1.9` and `@storybook/manager@8.2.0`:


https://github.com/user-attachments/assets/7373f678-1627-4a15-ac25-2edb33f643cf


### Reproduction link

https://stackblitz.com/~/github.com/sulco/storybook-addon-kit-repro?file=tsup.config.ts

### Reproduction steps

1. Go to the link above. It should automatically run `npm i && npm start`. It already includes the broken state.
2. After seeing the error, kill the process and run `npm i @storybook/manager@8.1.9`. Then run `npm start` and see that it works again.

Alternatively, you can reproduce it locally using https://github.com/storybookjs/addon-kit.
1. Clone the https://github.com/storybookjs/addon-kit
2. Update the storybook dependencies to the latest (e.g. running `npx npm-check -u`)
3. Try running `npm start`

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.4.1
    CPU: (11) arm64 Apple M3 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.12.0 - ~/.nvm/versions/node/v20.12.0/bin/node
    Yarn: 4.2.2 - ~/.nvm/versions/node/v20.12.0/bin/yarn
    npm: 10.5.0 - ~/.nvm/versions/node/v20.12.0/bin/npm <----- active
    pnpm: 9.3.0 - ~/Library/pnpm/pnpm
  Browsers:
    Chrome: 126.0.6478.128
    Safari: 17.4.1
  npmPackages:
    @storybook/addon-essentials: ^8.2.4 => 8.2.4 
    @storybook/addon-interactions: ^8.2.4 => 8.2.4 
    @storybook/addon-links: ^8.2.4 => 8.2.4 
    @storybook/blocks: ^8.2.4 => 8.2.4 
    @storybook/components: ^8.2.4 => 8.2.4 
    @storybook/core-events: ^8.2.4 => 8.2.4 
    @storybook/manager: 8.2 => 8.2.4 
    @storybook/manager-api: ^8.2.4 => 8.2.4 
    @storybook/preview: ^8.2.4 => 8.2.4 
    @storybook/preview-api: ^8.2.4 => 8.2.4 
    @storybook/react: ^8.2.4 => 8.2.4 
    @storybook/react-vite: ^8.2.4 => 8.2.4 
    @storybook/test: ^8.2.4 => 8.2.4 
    @storybook/theming: ^8.2.4 => 8.2.4 
    @storybook/types: ^8.2.4 => 8.2.4 
    storybook: ^8.2.4 => 8.2.4
```


### Additional context

_No response_

--- a/code/core/package.json
+++ b/code/core/package.json
@@ -142,6 +142,11 @@
       "import": "./dist/manager/globals-module-info.js",
       "require": "./dist/manager/globals-module-info.cjs"
     },
+    "./manager/globals": {
+      "types": "./dist/manager/globals.d.ts",
+      "import": "./dist/manager/globals.js",
+      "require": "./dist/manager/globals.cjs"
+    },
     "./preview/globals": {
       "types": "./dist/preview/globals.d.ts",
       "import": "./dist/preview/globals.js",
@@ -228,6 +233,9 @@
       ],
       "manager/globals-module-info": [
         "./dist/manager/globals-module-info.d.ts"
+      ],
+      "manager/globals": [
+        "./dist/manager/globals.d.ts"
       ],
       "preview/globals": [
         "./dist/preview/globals.d.ts"

--- a/code/core/scripts/entries.ts
+++ b/code/core/scripts/entries.ts
@@ -34,6 +34,7 @@ export const getEntries = (cwd: string) => {
     define('src/docs-tools/index.ts', ['browser', 'node'], true),
 
     define('src/manager/globals-module-info.ts', ['node'], true),
+    define('src/manager/globals.ts', ['node'], true),
     define('src/preview/globals.ts', ['node'], true),
   ];
 };

--- a/code/deprecated/manager/globals.js
+++ b/code/deprecated/manager/globals.js
@@ -1,0 +1,1 @@
+module.exports = require('storybook/internal/manager/globals');

--- a/code/deprecated/preview/globals.js
+++ b/code/deprecated/preview/globals.js
@@ -1,0 +1,1 @@
+module.exports = require('storybook/internal/preview/globals');

--- a/code/lib/cli/core/manager/globals.cjs
+++ b/code/lib/cli/core/manager/globals.cjs
@@ -1,0 +1,1 @@
+module.exports = require('@storybook/core/manager/globals');

--- a/code/lib/cli/core/manager/globals.d.ts
+++ b/code/lib/cli/core/manager/globals.d.ts
@@ -1,0 +1,2 @@
+export * from '@storybook/core/manager/globals';
+export type * from '@storybook/core/manager/globals';

--- a/code/lib/cli/core/manager/globals.js
+++ b/code/lib/cli/core/manager/globals.js
@@ -1,0 +1,1 @@
+export * from '@storybook/core/manager/globals';

--- a/code/lib/cli/package.json
+++ b/code/lib/cli/package.json
@@ -167,6 +167,11 @@
       "types": "./core/preview/globals.d.ts",
       "import": "./core/preview/globals.js",
       "require": "./core/preview/globals.cjs"
+    },
+    "./internal/manager/globals": {
+      "types": "./core/manager/globals.d.ts",
+      "import": "./core/manager/globals.js",
+      "require": "./core/manager/globals.cjs"
     }
   },
   "main": "dist/index.cjs",
@@ -218,6 +223,9 @@
       ],
       "internal/manager-errors": [
         "./core/manager-errors.d.ts"
+      ],
+      "internal/manager/globals": [
+        "./core/manager/globals.d.ts"
       ],
       "internal/manager/globals-module-info": [
         "./core/manager/globals-module-info.d.ts"

